raiden-dapp/src/store.ts

```diff
@@ -100,7 +101,7 @@ const store: StoreOptions<RootState> = {
     channels: (state: RootState) => (tokenAddress: string) => {
       let channels: RaidenChannel[] = [];
       const tokenChannels = state.channels[tokenAddress];
-      if (tokenChannels) {
+      if (tokenChannels && !isEmpty(tokenChannels)) {
```

Comment:
This doesn't solve the issue with the empty channels mapping. The issue is in `getters.token` map above, which uses the `state.channels`, not `getters.channels`. When `state.channels` contains something like `{ [token]: {} }`, `getters.tokens` will start returning emptyTokenModel without even the address, even though the address or token data is present in `state.tokens`. You need to either or both make `getters.tokens` behave well when finding a `[token]: {}` empty mapping in `state.channels` or ensure `updateChannels` mutation filter out these empty channels map from the mapping.

Reply:
I would prefer to have each reducer do the check, since you already mentioned that `{ [token]: {} }` is a valid structure.

Reply:
Then the `getters.tokens` map/reduce over it, instead of over `state.channels`, or the problem I mentioned would stay

Reply:
`getters.tokens` should also be updated to filter out tokens without channels.

Reply:
Oh, right. So that should be it. Thank you =)